test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -599,6 +599,57 @@ class test_ucp_sockaddr : public ucp_test {
         m_test_addr.set_port(1);
     }
 
+    void connect_and_fail_wireup(entity &e, ucp_ep_flags_t wait_ep_flags,
+                                 bool fail_msg_send)
+    {
+        start_listener(cb_type());
+
+        scoped_log_handler slh(wrap_errors_logger);
+        client_ep_connect();
+        if (!wait_for_server_ep(false)) {
+            UCS_TEST_SKIP_R("cannot connect to server");
+        }
+
+        if (fail_msg_send) {
+            /* Emulate failure of WIREUP MSG sending by setting the AM Bcopy
+             * function which always return EP_TIMEOUT error */
+            for (ucp_lane_index_t lane_idx = 0;
+                 lane_idx < ucp_ep_num_lanes(e.ep()); ++lane_idx) {
+                uct_ep_h uct_ep = e.ep()->uct_eps[lane_idx];
+
+                uct_ep->iface->ops.ep_am_bcopy =
+                        reinterpret_cast<uct_ep_am_bcopy_func_t>(
+                                ucs_empty_function_return_bc_ep_timeout);
+            }
+        }
+
+        while (!(e.ep()->flags & wait_ep_flags)) {
+            progress();
+        }
+
+        if (!fail_msg_send) {
+            /* Emulate failure of preparation of WIREUP MSG sending by setting
+             * the address getters to the function that always return error */
+            for (ucp_lane_index_t lane_idx = 0;
+                 lane_idx < ucp_ep_num_lanes(e.ep()); ++lane_idx) {
+                uct_ep_h uct_ep = e.ep()->uct_eps[lane_idx];
+
+                uct_ep->iface->ops.iface_get_device_address =
+                        reinterpret_cast<uct_iface_get_device_address_func_t>(
+                                ucs_empty_function_return_ep_timeout);
```

Comment:
maybe setting just one of these functions would be enough?

Reply:
done